app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -180,25 +167,34 @@ class BusinessEntityController @Inject()(
     implicit val eori: EORI = request.eoriNumber
     store.get[BusinessEntityJourney].flatMap {
       case Some(journey) =>
-        Future.successful(
-          Ok(
-            businessEntityCyaPage(
-              journey.eori.value.getOrElse(throw new IllegalStateException("missing eori")),
-              journey.contact.value.getOrElse(throw new IllegalStateException("missing contact details"))
-            )
-          )
-        )
+        val eori = journey.eori.value.getOrElse(handleMissingSessionData("EORI"))
+        val contactDetails = journey.contact.value.getOrElse(handleMissingSessionData("contact details"))
```

Comment:
Something to think about later. But it may be worth adding some validation function to a journey later. We seem to do this a lot (Don't bother as part of this though, it'll be a hairy refactor)

Reply:
yeah that will be done later 👍